pkg/v1/tkg/client/upgrade_cluster_clusterclass.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/clusterclient"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/log"
+)
+
+func (c *TkgClient) DoClassyClusterUpgrade(regionalClusterClient clusterclient.Client,
+	currentClusterClient clusterclient.Client, options *UpgradeClusterOptions) error {
+
+	kubernetesVersion := options.KubernetesVersion
+
+	log.Infof("Upgrading kubernetes cluster to `%v` version", kubernetesVersion)
+	patchJSONString := fmt.Sprintf(`{"spec": {"topology": {"version": "%v"}}}`, kubernetesVersion)
+
+	err := regionalClusterClient.PatchClusterObject(options.ClusterName, options.Namespace, patchJSONString)
+	if err != nil {
+		return errors.Wrap(err, "unable to patch kubernetes version to cluster")
+	}
+
+	log.Info("Waiting for kubernetes version to be updated for control plane nodes")
+	err = regionalClusterClient.WaitK8sVersionUpdateForCPNodes(options.ClusterName, options.Namespace, kubernetesVersion, currentClusterClient)
```

Comment:
Q: At first glance it looks like there's commonality between this and the version update that we do for non-cluster classes..https://github.com/vmware-tanzu/tanzu-framework/blob/2cb4eb34415f79e640fc0e5f99fe174b1faac1ea/pkg/v1/tkg/client/upgrade_cluster.go#L489-L504 - Is it possible to take these out into a helper function possibly?

Reply:
Because the legacy upgrade is more complicated and we also keep track of cluster upgrade status as well as patch some resources between the KCP and MD wait logic. 
Unfortunately because of the above reason it doesn't make sense to create a common helper function for this wait logic.